lib/galaxy/web/proxy/__init__.py

```diff
@@ -243,20 +244,16 @@ def handle_requests(self, authentication, proxy_requests, route_name, container_
             'ContainerIds': container_ids,
         }
 
-        req = urllib2.Request(self.api_url)
-        req.add_header('Content-Type', 'application/json')
-
         # Sometimes it takes our poor little proxy a second or two to get
         # going, so if this fails, re-call ourselves with an increased timeout.
         try:
-            urllib2.urlopen(req, json.dumps(values))
-        except urllib2.URLError as err:
-            log.debug(err)
+            requests.get(self.api_url, headers={'Content-Type': 'application/json'}, data=json.dumps(values))
+        except requests.exceptions.ConnectionError as err:
+            log.exception(err)
             if sleep > 5:
                 excp = "Could not contact proxy after %s seconds" % sum(range(sleep + 1))
                 raise Exception(excp)
             time.sleep(sleep)
-            self.handle_requests(authentication, proxy_requests, route_name, container_ids, sleep=sleep + 1)
-        pass
+            self.handle_requests(authentication, proxy_requests, route_name, container_ids, container_interface, sleep=sleep + 1)
```

Comment:
`./lib/galaxy/web/proxy/__init__.py:257:93: F821 undefined name 'container_interface'`

You need to remove `container_interface, ` from this call, the parameter was added in release 17.05: https://github.com/galaxyproject/galaxy/pull/3863/files#diff-89850e7b31329e95fcf2e0475b5769d5L302